extensions/ssh/runtime/src/main/java/org/apache/camel/quarkus/component/ssh/runtime/SubstituteSecurityUtils.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.ssh.runtime;
+
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import org.apache.sshd.common.util.buffer.Buffer;
+import org.apache.sshd.common.util.security.SecurityUtils;
+
+/**
+ * We're substituting those offending methods that would require the presence of
+ * net.i2p.crypto:eddsa library which is not supported by Camel SSH component
```

Comment:
Is this documented somewhere? I am not finding any mention of eddsa on https://camel.apache.org/components/latest/ssh-component.html Shouldn't we file a documentation issue?

Reply:
I think I did not explain properly in the comment. The `camel-ssh` depends on `sshd-core` library which has 2 optional security providers, `bouncycastle` and `eddsa`. In `camel-ssh` we chose to use `bouncycastle`. I will rephrase the comment to highlight that.

Reply:
> I think I did not explain properly in the comment. The `camel-ssh` depends on `sshd-core` library which has 2 optional security providers, `bouncycastle` and `eddsa`. In `camel-ssh` we chose to use `bouncycastle`.

Thanks for the clarification. I still think that this is a piece of info interesting for the end users and it should ideally be present on https://camel.apache.org/components/latest/ssh-component.html . Filing a documentation issue would be a way not forget about that :)

Reply:
Okey, I'll take care!

Reply:
For the record: https://issues.apache.org/jira/browse/CAMEL-15550